specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2022-10-01-preview/settings.json

```diff
@@ -0,0 +1,313 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeSettingParameter"
+          },
+          {
+            "$ref": "#/parameters/settingTypeParameter"
+          },
+          {
+            "$ref": "common-types.json#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/Setting"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Settings"
+        ],
+        "operationId": "Settings_DeleteByScope",
+        "description": "Delete a setting within the given scope.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "SettingDeleteByScope": {
+            "$ref": "./examples/setting-delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/scopeSettingParameter"
+          },
+          {
+            "$ref": "#/parameters/settingTypeParameter"
+          },
+          {
+            "$ref": "common-types.json#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded."
+          },
+          "204": {
+            "description": "OK. The request has succeeded."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Setting": {
+      "description": "Setting definition.",
+      "discriminator": "kind",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "common-types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "Specifies the kind of settings.",
+          "enum": [
+            "taginheritance"
+          ],
+          "x-ms-enum": {
+            "name": "SettingsKind",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "kind"
+      ]
+    },
+    "TagInheritanceSetting": {
+      "description": "Tag Inheritance Setting definition.",
+      "type": "object",
+      "x-ms-discriminator-value": "taginheritance",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Setting"
+        }
+      ],
+      "properties": {
+        "properties": {
```

Comment:
See the use of "x-ms-client-flatten" to avoid undesired nested level of properties. See: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten).

Reply:
will keep it as-is for now.